rust/worker/src/execution/operators/merge_knn_results.rs

```diff
@@ -95,7 +96,25 @@ impl Operator<MergeKnnResultsOperatorInput, MergeKnnResultsOperatorOutput>
                     if input.include_vectors {
                         hnsw_result_vectors = Some(Vec::new());
                     }
-
+                    // Prefetch data.
+                    // Note: This will be moved to an operator.
```

Comment:
Would prefer we just stack a PR and do the right thing. I don't see value in merging this in haste unfinished.

Reply:
I have created a stack. Will abandon this PR in favor of that. My intent behind doing this change was to quickly measure some gains in staging by merging this. This change is incrementally better over the current version in main but I am ok with removing it. Otherwise, we have to do a bunch of things that needs to be done before we can leverage - dedup, NAC, operators, runtime